### DISCUSSION TITLE
Why probability defaults to 0.5 ?

### DISCUSSION THREAD

**Discussion by trya2l** [iver56/audiomentations#366] [Q&A]
Hello,

I'm just getting started with audiomentations, but I was wondering if there is a rational for defaulting probability to 0.5 ?

It seems a bit counterintuitive to me, maybe it's because I use it in a deterministic way ?

Anyway, thanks for this lib. It's awesome

**Comment by iver56** [comment]
Good question. I guess this was inspired by what albumentations was 6 years ago. You are not the first person to question this decision. Does it make more sense to default to p=1.0? What are the pros and cons?

**Reply by trya2l** [reply]
Ok, I see. So this is my beginner's opinion, maybe I'm overlooking some aspects, but to cite just my case, I don't use it randomly, which forces me to add `p=1` everywhere. Apart from that, I'm not sure how people generally use it, but I've seen an example where it is applied only on `Compose`.

```python


# Create audio augmentations
audio_augmentations = Compose([
    AddGaussianSNR(min_snr_db=10, max_snr_db=20),
    Gain(min_gain_db=-6, max_gain_db=6),
    GainTransition(min_gain_db=-6, max_gain_db=6, min_duration=0.01, max_duration=0.3, duration_unit="fraction"),
    ClippingDistortion(min_percentile_threshold=0, max_percentile_threshold=30, p=0.5),
    TimeStretch(min_rate=0.8, max_rate=1.2),
    PitchShift(min_semitones=-4, max_semitones=4),
], p=0.8, shuffle=True)
```

https://huggingface.co/SleepyJesse/ai_music_detection_large_60s/blob/main/ai_music_detection_new_large_60.ipynb
